lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs

```diff
@@ -47,11 +47,20 @@ private HadoopFileSystem(HdfsCommandRunner commandRunner)
         }
 
         /// <summary>
-        /// The Prefix used for URIs on this FileSystem.
+        /// CReate Uri from a given file name
+        /// If the file doesn't contain the preFIx, add the prefix in front
+        /// otherwise, use it directly
         /// </summary>
-        public string UriPrefix
+        /// <param name="path"></param>
+        /// <returns></returns>
+        public Uri CreateUriForPath(string path)
         {
-            get { return _uriPrefix; }
+            var fullName = path;
+            if (!path.StartsWith(_uriPrefix))
```

Comment:
This is brittle. Maybe we should convert the string to a Uri and then check for the protocol / authority to be equal?